adblock/rules.go

```diff
@@ -497,12 +503,11 @@ func (n *ruleNode) matchChildren(ctx *matchContext, url []byte, rq *Request) (
 		return -1, nil
 	}
 	if len(url) == 0 && len(n.Children) == 0 {
-		domains := 0
+		domains, matchedDomains := matchOptsDomains(n.Opts, rq.OriginDomain)
```

Comment:
Note that we need to look at all domain options together because they can be specified in separate rule lines. An example from the actual `easylist.txt` is:

```
||s3.amazonaws.com^$domain=imgclick.net
||s3.amazonaws.com^$domain=gaybeeg.info
```